.pipelines/cni/lsg/lsg-cni-intergration-template.yaml

```diff
@@ -149,17 +181,71 @@ stages:
-          sub: $(BUILD_VALIDATIONS_SERVICE_CONNECTION)
-          clusterName: ${{ parameters.clusterName }}-$(commitID)
-          os: ${{ parameters.os }}
-          dependsOn: recover
-          datapath: true
-          dns: true
-          portforward: true
-          service: true
-          hostport: true
+      - job: cni_tests
+        displayName: "Cilium Test"
+        dependsOn: recover
+        condition: and( succeeded(), ${{ contains(parameters.name, 'cilium') }} )
+        steps:
+          - script: |
+              echo "install cilium CLI"
+              if [[ ${CILIUM_VERSION_TAG} =~ ^1.1[1-3].[0-9]{1,2} ]]; then
+                echo "Cilium Agent Version ${BASH_REMATCH[0]}"
+                CILIUM_CLI_VERSION=$(curl -s https://raw.githubusercontent.com/cilium/cilium-cli/main/stable-v0.14.txt)
+              else
+                echo "Cilium Agent Version ${CILIUM_VERSION_TAG}"
+                CILIUM_CLI_VERSION=$(curl -s https://raw.githubusercontent.com/cilium/cilium-cli/master/stable.txt)
+              fi
+              CLI_ARCH=amd64
+              curl -L --fail --remote-name-all https://github.com/cilium/cilium-cli/releases/download/${CILIUM_CLI_VERSION}/cilium-linux-${CLI_ARCH}.tar.gz{,.sha256sum}
+              sha256sum --check cilium-linux-${CLI_ARCH}.tar.gz.sha256sum
+              sudo tar xzvfC cilium-linux-${CLI_ARCH}.tar.gz /usr/local/bin
+              rm cilium-linux-${CLI_ARCH}.tar.gz{,.sha256sum}
+              cilium status
+              cilium version
+            name: "InstallCiliumCli"
+            displayName: "Install Cilium CLI"
+          - task: AzureCLI@1
+            inputs:
+              azureSubscription: $(BUILD_VALIDATIONS_SERVICE_CONNECTION)
+              scriptLocation: "inlineScript"
+              scriptType: "bash"
+              addSpnToEnvironment: true
+              inlineScript: |
+                set -ex
+                make -C ./hack/aks set-kubeconf AZCLI=az CLUSTER=${{ parameters.clusterName }}-$(commitID)
+            name: "GetCluster"
+            displayName: "Get AKS Cluster"
+          - script: |
+              kubectl delete ns load-test
+              cilium connectivity test --connect-timeout 4s --request-timeout 30s --test '!pod-to-pod-encryption,!node-to-node-encryption'
+            retryCountOnTaskFailure: 6
+            name: "CiliumConnectivityTests"
+            displayName: "Run Cilium Connectivity Tests"
+      - ${{ if contains(parameters.name, 'cilium') }}:
+        - template: ../k8s-e2e/k8s-e2e-job-template.yaml
+          parameters:
+            sub: $(BUILD_VALIDATIONS_SERVICE_CONNECTION)
+            clusterName: ${{ parameters.clusterName }}-$(commitID)
+            os: ${{ parameters.os }}
+            cni: ${{ parameters.cni }}
+            dependsOn: cni_tests
+            datapath: true
+            dns: true
+            portforward: true
+            service: true
+      - ${{ if contains(parameters.name, 'linux') }}:
```

Comment:
Would it be better to have a condition on cni (given we do change stage name in future)?

Reply:
I agree. I think if we want to continue increasing this matrix into dulastack, which would break the if condition on CNI, we would make another template.